src/object-store/src/property.hpp

```diff
@@ -23,25 +23,15 @@
 
 namespace realm {
     enum PropertyType {
-        /** Integer type: NSInteger, int, long, Int (Swift) */
         PropertyTypeInt    = 0,
-        /** Boolean type: BOOL, bool, Bool (Swift) */
         PropertyTypeBool   = 1,
-        /** Float type: CGFloat (32bit), float, Float (Swift) */
         PropertyTypeFloat  = 9,
-        /** Double type: CGFloat (64bit), double, Double (Swift) */
         PropertyTypeDouble = 10,
-        /** String type: NSString, String (Swift) */
         PropertyTypeString = 2,
-        /** Data type: NSData */
         PropertyTypeData   = 4,
-        /** Any type: id, **not supported in Swift** */
-        PropertyTypeAny    = 6,
-        /** Date type: NSDate */
-        PropertyTypeDate   = 7,
-        /** Object type. See [Realm Models](http://realm.io/docs/cocoa/latest/#models) */
+        PropertyTypeAny    = 6, // deprecated and will be removed in the future
+        PropertyTypeDate   = 8,
```

Comment:
`PropertyTypeDate` changed here from `7` to `8`. I'm assuming that's on purpose because of the format change, but then how does the migration code in `js_realm.hpp` work correctly when it's checking against `8`?